src/xtopcom/xca_auth/xca_auth.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2017-2021 Telos Foundation & contributors
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#pragma once
+
+#include "xbase/xns_macro.h"
+#include <stdint.h>
+#include <string.h>
+#include "openssl/evp.h"
+#include "openssl/x509.h"
```

Comment:
Since the `xtop_ca_auth` struct's public api has only three: `add_root_cert` , `verify_leaf_cert`, `get_cert_expiry_time`, none of which contains lib `openssl` 's data structures. I strongly advise moving these two header files into an inner implementation header file to avoid leaking these details to any unit that include this file, in this case, the `xvm/xsystem_contracts/consortium/xnode_manage_contract.cpp`.

I might to do the following steps:

1. Create a new folder `src/xtopcom/xca_auth/xca_auth_impl/`
2. Create new files `xca_auth_impl.h` and `xca_auth_impl.cpp` in the new folder, containing the class `xtop_ca_auth_impl`.
3. Move the contents of these two files into the new implementation files, but only retain the above three APIs. This way, we can delete the include OpenSSL lines.
4. This `xtop_ca_auth` class will hold a pointer to `xtop_ca_auth_impl`. Use forward declaration in this header file and only include `xca_auth_impl.h` in `xca_auth.cpp`.